### DISCUSSION TITLE
Welcome to STAC Discussions!

### DISCUSSION THREAD

**Discussion by Xylopyrographer** [Xylopyrographer/STAC#2] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mqshaw** [comment]
The service Sunday night went great and STAC was awesome !!

A couple of things that I thought of that I would share with you

1)	During my testing ( at home ) I noticed that the ATOM was not always getting a response back in time from the web server. This was resulting in a periodic purple X to appear to on the display and then disappear.  It is somewhat sporadic, and I am not sure if it is a result of network traffic of my server response. So, to prevent this from happening, I developed a tracking function that monitors the last 8 responses and only displays the X if multiple “non-responses” occur. I have  this on a local git branch, but think that you might like to integrate this feature.

2)	Our V-60HD is connected to a router using DHCP. Using the current user interface one must reset the STAC to enter the new server address. Here is what I was thinking ... what about having a simple web server running on the device ( maybe when it cannot connect to the V-60HD ). Maybe show a scrolling message on the STAC that says "Cannot connect to V-60HD, STAC IP address is : XXX.XXX.XXX.XXX"  By navigating to the embedded web server using a device on the network, one can reset the IP address of the V-60HD.  This would prevent having to reconnect the device to the network using the security credentials every time.

**Reply by Xylopyrographer** [reply]
So glad to hear the STAC was a hit. Truly appreciate the feedback.

**On point 1.**
Pretty sure this is due to network turn around. On my test system at home and in real life at the church, I've only seen the X when I've induced a delay in the reply. Having said that, I like the concept.

This ties in a bit to my thoughts on having an advanced configuration tab on the web page server by the STAC. In this specific case, being able to tweak the `ST_POLL_INTERVAL` might be a way to optimize on a specific network. Likewise tweaking a "Number of missed responses to a poll request error" variable could be included.

 FWIW, at church we have a dedicated Internet connection just for the streaming kit. On that network we pump out the live stream feed, run an NDI feed from the FOH computer to the "Studio" computer, run five STACs, do the control of a PTZ camera and have a few other devices (mobile devices & another couple of computers) for monitoring things. So far have not seen a bandwidth issue.

**On point 2.**
I really think best practice is to configure the V-60HD with a static IP address. Possible to chat with the person who runs the network and dialog on that?

In lieu of that, and thinking about the pre-service workflow, the camera operator(s) would have to contact the person running the V60-HD have them retrieve the IP address and then set that into the STAC.

Thoughts?

**Reply by mqshaw** [reply]
On Point 1
I agree, most likely.  How do you want to handle this, should I submit a pull request?
The one thing that I am puzzled by is why the STAC seems to to out top lunch when I kill the web service.  I would like to respond with the X sooner, but for some reason it seems to take 32 seconds before that section of the code is hit a second time to be able to change to the X.

On Point 2
Yes, I would rather use statics IP also. I am the person that manages to the IT infrastructure also, so that not a problem.  The problem that i have had is the router that we are using does not seem to manage static IPs very well.  And we were having issues with other devices not being able to connect to the network.  I can take a look again, maybe it has been fixed with a subsequent FW update. Probably time to check again.

**Comment by simardmaryd** [comment]
Awesome and great to know that. Now i can make changes to the source code of [asphalt 9 mod apk](https://apkholder.com/asphalt-9-mod-apk/) and share it with others to spread happiness. All thanks to this community.
